core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka.scaladsl
+
+import scala.concurrent.Future
+import akka.Done
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import akka.kafka.internal.ConsumerStage
+import akka.NotUsed
+import akka.dispatch.ExecutionContexts
+import scala.concurrent.ExecutionContext
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import akka.kafka.internal.ConsumerStage.CommittableOffsetBatchImpl
+import akka.kafka.internal.PlainConsumerStage
+import akka.kafka.internal.CommittableConsumerStage
+
+object Consumer {
+
+  trait Message[K, V] {
+    def key: K
+    def value: V
+    def partitionOffset: PartitionOffset
+  }
+
+  final case class CommittableMessage[K, V](
+      key: K,
+      value: V,
+      committableOffset: CommittableOffset
+  ) extends Message[K, V] {
+
+    override def partitionOffset: PartitionOffset = committableOffset.partitionOffset
+  }
+
+  /**
+   * This interface might move into `akka.stream`
+   */
+  trait Committable {
+    def commit(): Future[Done]
+  }
+
+  trait CommittableOffset extends Committable {
+    def partitionOffset: PartitionOffset
+  }
+
+  final case class PartitionOffset(key: ClientTopicPartition, offset: Long)
+
+  final case class ClientTopicPartition(
+    clientId: String,
+    topic: String,
+    partition: Int
+  )
+
+  object CommittableOffsetBatch {
+    val empty: CommittableOffsetBatch = new CommittableOffsetBatchImpl(Map.empty, Map.empty)
+  }
+
+  trait CommittableOffsetBatch extends Committable {
+    def updated(offset: CommittableOffset): CommittableOffsetBatch
+
+    def getOffset(key: ClientTopicPartition): Option[Long]
```

Comment:
@13h3r with these two methods the user should be able to implement whatever batching she likes, e.g. `updatedWithLatest` can be implemented.